### PR TITLE
Elastic Endpoint 8.5.0 yaml

### PR DESCRIPTION
**Pull request by lrishi**

### CODE REVIEW

**Review by nick-alayil — APPROVED** [review]
LGTM

**Comment by nick-alayil** [comment]
Ah. Too quick to approve ;-). Just noticed that there is a new addition in the default yaml from elastic-agent - added a new toleration for running agent on k8s control-plane nodes. This don't affect endpoint in anyway but just to make sure our yaml is a superset of existing default yaml. 

```
        - key: node-role.kubernetes.io/control-plane
          effect: NoSchedule
```

**Review by nick-alayil — APPROVED** [review]
Perfect!

File: releases/8.5.0/kubernetes/deploy/elastic-defend.yaml [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+      - events
+      - pods
+      - services
+      - configmaps
+      # Needed for cloudbeat
+      - serviceaccounts
+      - persistentvolumes
+      - persistentvolumeclaims
+    verbs: ["get", "list", "watch"]
+  # Enable this rule only if planing to use kubernetes_secrets provider
```
Comment (jrmolin) [inline releases/8.5.0/kubernetes/deploy/elastic-defend.yaml:248]:
```suggestion
  # Enable this rule only if planning to use kubernetes_secrets provider
```

**Review by jrmolin — APPROVED** [review]
one spelling fix. honestly, it's too much yaml for my :eyes: 

I never really got into k8s for this reason

**Review by qcorporation — APPROVED** [review]

**Comment by qcorporation** [comment]
@benironside we need to update the K8s onboarding docs to include this new YAML file fetch (https://github.com/elastic/endpoint/tree/main/releases/8.5.0/kubernetes/deploy)

**Pull request merged**